nginx-asg.tf

```diff
@@ -0,0 +1,194 @@
+data "aws_ami" "bgdc_nginx_latest" {
+  most_recent = true
+
+  filter {
+    name   = "name"
+    values = ["bgdc-nginx-ami-main-*"]
+  }
+
+  filter {
+    name   = "virtualization-type"
+    values = ["hvm"]
+  }
+
+  owners = [local.bgdc_account.test]
+}
+
+resource "aws_launch_configuration" "nginx_conf" {
+  name            = "dwx_bgdc_nginx"
+  image_id        = data.aws_ami.bgdc_nginx_latest.id
+  instance_type   = "t2.medium"
+  security_groups = [aws_security_group.nginx-bgdc-dwx.id]
+  iam_instance_profile = aws_iam_instance_profile.dwx_bgdc_nginx_instance_profile.arn
+ 
+  
+  user_data = templatefile("bootstrapBgdcDwxNginx.template", { BgdcDwxListener = 400, BgdcDwxNginxDns = "localhost", BgdcDwxNginxPort = 400 })
+
+  root_block_device {
+    volume_type           = "gp3"
+  }
+
+  lifecycle {
+    create_before_destroy = false
+  }
+
+}
+
+resource "aws_autoscaling_group" "nginx_asg" {
+  name                      = "dwx_bgdc_nginx"
+  max_size                  = 2
+  min_size                  = 2
+  health_check_grace_period = 300
+  health_check_type         = "ELB"
+  desired_capacity          = 2
+  force_delete              = true
+  launch_configuration      = aws_launch_configuration.nginx_conf.name
+  vpc_zone_identifier       = data.terraform_remote_state.internal_compute.outputs.bgdc_subnet.ids
+  target_group_arns         = [aws_lb_target_group.dwx_bdgc_nginx_nlb_tg.arn]
+
+  lifecycle {
+    create_before_destroy = true
+  }
+
+  tag {
+    key                 = "Name"
+    value               = "dwx-bgdc-nginx-instances"
+    propagate_at_launch = true
+  }
+
+}
+
+
+resource "aws_lb" "dwx_bdgc_nginx_nlb" {
+  name               = "dwx-bgdc-nginx-nlb"
```

Comment:
This name has dashes where as other resources have names with underscores in. Can we keep this consistent so its easier for when people refer to remote states